src/Raven.Server/Documents/Subscriptions/LiveSubscriptionPerformanceCollector.cs

```diff
@@ -0,0 +1,304 @@
+// -----------------------------------------------------------------------
+//  <copyright file="LiveSubscriptionPerformanceCollector.cs" company="Hibernating Rhinos LTD">
+//      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Server.Documents.Subscriptions.Stats;
+using Raven.Server.Documents.TcpHandlers;
+using Raven.Server.Json;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils.Stats;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Subscriptions
+{
+    public class LiveSubscriptionPerformanceCollector : DatabaseAwareLivePerformanceCollector<SubscriptionTaskPerformanceStats>
+    {
+        private readonly ConcurrentDictionary<string, SubscriptionAndPerformanceConnectionStatsList> _perSubscriptionConnectionStats
+            = new ConcurrentDictionary<string, SubscriptionAndPerformanceConnectionStatsList>();
+        
+        private readonly ConcurrentDictionary<string, SubscriptionAndPerformanceBatchStatsList> _perSubscriptionBatchStats
+            = new ConcurrentDictionary<string, SubscriptionAndPerformanceBatchStatsList>();
+        
+        public LiveSubscriptionPerformanceCollector(DocumentDatabase database) : base(database)
+        {
+            var initialStats = PrepareInitialPerformanceStats().ToList();
+            if (initialStats.Count > 0)
+            {
+                Stats.Enqueue(initialStats);
+            }
+            
+            Start();
+        }
+
+        // Upon collector init only: get history stats + in progress stats (to be sent to ws)
```

Comment:
I think this obvious on method name: preapre *initial*.

Reply:
done